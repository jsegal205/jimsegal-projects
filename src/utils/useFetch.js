import { useState, useEffect } from "react";
import axios from "axios";

// custom hook for performing GET request
const useFetch = (url, initialValue) => {
  const [data, setData] = useState(initialValue);
  const [loading, setLoading] = useState(true);
  useEffect(() => {
    const fetchData = async () => {
      try {
        setLoading(true);
        const response = await axios.get(url);
        setData(response.data);
      } catch ({ response }) {
        setData({ status: response.status });
      } finally {
        setLoading(false);
      }
    };
    fetchData();
  }, [url]);
  return { loading, data };
};

export default useFetch;
