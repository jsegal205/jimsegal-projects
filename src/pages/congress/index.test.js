import React from "react";
import { BrowserRouter as Router } from "react-router-dom";

import useFetch from "../../utils/useFetch";
import Congress from "./index";
import { render } from "@testing-library/react";
import { toMatchDiffSnapshot } from "snapshot-diff";

expect.extend({ toMatchDiffSnapshot });

jest.mock("../../utils/useFetch", () => jest.fn());

const returnData = {
  session: 116,
  house: {
    age: {
      average: {
        all: 58.52,
        democrat: 59.39,
        female: 58.93,
        male: 58.4,
        republican: 57.62,
      },
      distribution: {
        "50": {
          total: 0,
          D: 0,
          R: 0,
          M: 0,
          F: 0,
        },
      },
      oldest: {
        age: 87,
        date_of_birth: "1933-06-09",
        full_name: "Don Young",
        gender: "M",
        party: "R",
        state: "AK",
      },
      youngest: {
        age: 30,
        date_of_birth: "1989-10-13",
        full_name: "Alexandria Ocasio-Cortez",
        gender: "F",
        party: "D",
        state: "NY",
      },
    },
    gender: {
      total: 449,
      men: 343,
      percentMen: 76.39,
      percentWomen: 23.61,
      women: 106,
    },
    party: {
      D: {
        total: 239,
        men: 148,
        percentMen: 61.92,
        percentWomen: 38.08,
        women: 91,
      },
      R: {
        total: 209,
        men: 194,
        percentMen: 92.82,
        percentWomen: 7.18,
        women: 15,
      },
    },
  },
  senate: {
    age: {
      average: {
        all: 63.89,
        democrat: 63.87,
        female: 62.08,
        male: 64.52,
        republican: 63.43,
      },
      oldest: {
        age: 87,
        date_of_birth: "1933-06-22",
        full_name: "Dianne Feinstein",
        gender: "F",
        party: "D",
        state: "CA",
      },
      youngest: {
        age: 40,
        date_of_birth: "1979-12-31",
        full_name: "Joshua Hawley",
        gender: "M",
        party: "R",
        state: "MO",
      },
    },
    gender: {
      total: 101,
      men: 75,
      percentMen: 74.26,
      percentWomen: 25.74,
      women: 26,
    },
    party: {
      D: {
        total: 45,
        men: 28,
        percentMen: 62.22,
        percentWomen: 37.78,
        women: 17,
      },
      R: {
        total: 54,
        men: 45,
        percentMen: 83.33,
        percentWomen: 16.67,
        women: 9,
      },
    },
  },
};

describe("Recipes Component", () => {
  describe("when `useFetch` is awaiting promise to resolve", () => {
    it("displays `Loading...`", () => {
      useFetch.mockReturnValue({
        loading: true,
        data: null,
      });

      const { container } = render(
        <Router>
          <Congress />
        </Router>
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("when `useFetch` returns data", () => {
    it("displays data", () => {
      useFetch.mockReturnValue({
        loading: false,
        data: returnData,
      });

      const { container } = render(
        <Router>
          <Congress />
        </Router>
      );
      expect(container).toMatchSnapshot();
    });

    describe("when 500 status returned", () => {
      it("displays error messaging", () => {
        useFetch.mockReturnValue({
          loading: false,
          data: {
            error: "server likely on fire",
            status: 500,
          },
        });

        const { container } = render(
          <Router>
            <Congress />
          </Router>
        );
        expect(container).toMatchSnapshot();
      });
    });
  });
});
