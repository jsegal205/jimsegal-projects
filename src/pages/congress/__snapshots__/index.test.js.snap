// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes Component when \`useFetch\` is awaiting promise to resolve displays \`Loading...\` 1`] = `
<div>
  <section>
    <h2
      class="congress-header"
    >
      Congressional Information
    </h2>
    <section
      class="congress-header"
    >
      Loading...
    </section>
  </section>
</div>
`;

exports[`Recipes Component when \`useFetch\` returns data displays data 1`] = `
<div>
  <section>
    <h2
      class="congress-header"
    >
      Congressional Information
    </h2>
    <div
      class="congress-subheader"
    >
      Session #
      116
    </div>
    <div
      class="congress"
    >
      <article>
        <h3>
          House
        </h3>
        <hr />
        <section
          class="chamber"
        >
          <section
            class="chamber-chart"
          >
            <div
              class="recharts-wrapper"
              style="position: relative; cursor: default; width: 921.6px; height: 400px;"
            >
              <svg
                class="recharts-surface"
                height="400"
                version="1.1"
                viewBox="0 0 921.6 400"
                width="921.6"
              >
                <defs>
                  <clippath
                    id="recharts1-clip"
                  >
                    <rect
                      height="330"
                      width="841.6"
                      x="70"
                      y="20"
                    />
                  </clippath>
                </defs>
                <g
                  class="recharts-cartesian-grid"
                >
                  <g
                    class="recharts-cartesian-grid-horizontal"
                  >
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="70"
                      x2="911.6"
                      y="20"
                      y1="350"
                      y2="350"
                    />
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="70"
                      x2="911.6"
                      y="20"
                      y1="267.5"
                      y2="267.5"
                    />
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="70"
                      x2="911.6"
                      y="20"
                      y1="185"
                      y2="185"
                    />
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="70"
                      x2="911.6"
                      y="20"
                      y1="102.5"
                      y2="102.5"
                    />
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="70"
                      x2="911.6"
                      y="20"
                      y1="20"
                      y2="20"
                    />
                  </g>
                  <g
                    class="recharts-cartesian-grid-vertical"
                  >
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="490.8"
                      x2="490.8"
                      y="20"
                      y1="20"
                      y2="350"
                    />
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="70"
                      x2="70"
                      y="20"
                      y1="20"
                      y2="350"
                    />
                    <line
                      fill="none"
                      height="330"
                      stroke="#ccc"
                      stroke-dasharray="3 3"
                      width="841.6"
                      x="70"
                      x1="911.6"
                      x2="911.6"
                      y="20"
                      y1="20"
                      y2="350"
                    />
                  </g>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
                >
                  <line
                    class="recharts-cartesian-axis-line"
                    fill="none"
                    height="30"
                    stroke="#666"
                    width="841.6"
                    x="70"
                    x1="70"
                    x2="911.6"
                    y="350"
                    y1="350"
                    y2="350"
                  />
                  <g
                    class="recharts-cartesian-axis-ticks"
                  >
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <line
                        class="recharts-cartesian-axis-tick-line"
                        fill="none"
                        height="30"
                        stroke="#666"
                        width="841.6"
                        x="70"
                        x1="490.8"
                        x2="490.8"
                        y="350"
                        y1="356"
                        y2="350"
                      />
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="30"
                        stroke="none"
                        text-anchor="middle"
                        width="841.6"
                        x="490.8"
                        y="358"
                      >
                        <tspan
                          dy="0.71em"
                          x="490.8"
                        >
                          50
                        </tspan>
                      </text>
                    </g>
                  </g>
                  <text
                    class="recharts-text recharts-label"
                    text-anchor="middle"
                    x="490.8"
                    y="385"
                  >
                    <tspan
                      dy="0.71em"
                      x="490.8"
                    >
                      Age
                    </tspan>
                  </text>
                </g>
                <g
                  class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
                >
                  <line
                    class="recharts-cartesian-axis-line"
                    fill="none"
                    height="330"
                    stroke="#666"
                    width="60"
                    x="10"
                    x1="70"
                    x2="70"
                    y="20"
                    y1="20"
                    y2="350"
                  />
                  <g
                    class="recharts-cartesian-axis-ticks"
                  >
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <line
                        class="recharts-cartesian-axis-tick-line"
                        fill="none"
                        height="330"
                        stroke="#666"
                        width="60"
                        x="10"
                        x1="64"
                        x2="70"
                        y="20"
                        y1="350"
                        y2="350"
                      />
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="330"
                        stroke="none"
                        text-anchor="end"
                        width="60"
                        x="62"
                        y="350"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          0
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <line
                        class="recharts-cartesian-axis-tick-line"
                        fill="none"
                        height="330"
                        stroke="#666"
                        width="60"
                        x="10"
                        x1="64"
                        x2="70"
                        y="20"
                        y1="267.5"
                        y2="267.5"
                      />
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="330"
                        stroke="none"
                        text-anchor="end"
                        width="60"
                        x="62"
                        y="267.5"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          1
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <line
                        class="recharts-cartesian-axis-tick-line"
                        fill="none"
                        height="330"
                        stroke="#666"
                        width="60"
                        x="10"
                        x1="64"
                        x2="70"
                        y="20"
                        y1="185"
                        y2="185"
                      />
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="330"
                        stroke="none"
                        text-anchor="end"
                        width="60"
                        x="62"
                        y="185"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          2
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <line
                        class="recharts-cartesian-axis-tick-line"
                        fill="none"
                        height="330"
                        stroke="#666"
                        width="60"
                        x="10"
                        x1="64"
                        x2="70"
                        y="20"
                        y1="102.5"
                        y2="102.5"
                      />
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="330"
                        stroke="none"
                        text-anchor="end"
                        width="60"
                        x="62"
                        y="102.5"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          3
                        </tspan>
                      </text>
                    </g>
                    <g
                      class="recharts-layer recharts-cartesian-axis-tick"
                    >
                      <line
                        class="recharts-cartesian-axis-tick-line"
                        fill="none"
                        height="330"
                        stroke="#666"
                        width="60"
                        x="10"
                        x1="64"
                        x2="70"
                        y="20"
                        y1="20"
                        y2="20"
                      />
                      <text
                        class="recharts-text recharts-cartesian-axis-tick-value"
                        fill="#666"
                        height="330"
                        stroke="none"
                        text-anchor="end"
                        width="60"
                        x="62"
                        y="20"
                      >
                        <tspan
                          dy="0.355em"
                          x="62"
                        >
                          4
                        </tspan>
                      </text>
                    </g>
                  </g>
                </g>
                <g
                  class="recharts-layer recharts-bar"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangles"
                  >
                    <g
                      class="recharts-layer"
                    >
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                    </g>
                  </g>
                </g>
                <g
                  class="recharts-layer recharts-bar"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangles"
                  >
                    <g
                      class="recharts-layer"
                    >
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                    </g>
                  </g>
                </g>
                <g
                  class="recharts-layer recharts-bar"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangles"
                  >
                    <g
                      class="recharts-layer"
                    >
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                    </g>
                  </g>
                </g>
                <g
                  class="recharts-layer recharts-bar"
                >
                  <g
                    class="recharts-layer recharts-bar-rectangles"
                  >
                    <g
                      class="recharts-layer"
                    >
                      <g
                        class="recharts-layer recharts-bar-rectangle"
                      />
                    </g>
                  </g>
                </g>
              </svg>
              <div
                class="recharts-legend-wrapper"
                style="position: absolute; width: 901.6px; height: auto; left: 10px; top: 20px;"
              >
                <ul
                  class="recharts-default-legend"
                  style="padding: 0px; margin: 0px; text-align: center;"
                >
                  <li
                    class="recharts-legend-item legend-item-0"
                    style="display: inline-block; margin-right: 10px;"
                  >
                    <svg
                      class="recharts-surface"
                      height="14"
                      style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                      version="1.1"
                      viewBox="0 0 32 32"
                      width="14"
                    >
                      <path
                        class="recharts-legend-icon"
                        d="M0,4h32v24h-32z"
                        fill="#0015BC"
                        stroke="none"
                      />
                    </svg>
                    <span
                      class="recharts-legend-item-text"
                    >
                      Democrat
                    </span>
                  </li>
                  <li
                    class="recharts-legend-item legend-item-1"
                    style="display: inline-block; margin-right: 10px;"
                  >
                    <svg
                      class="recharts-surface"
                      height="14"
                      style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                      version="1.1"
                      viewBox="0 0 32 32"
                      width="14"
                    >
                      <path
                        class="recharts-legend-icon"
                        d="M0,4h32v24h-32z"
                        fill="#E9141D"
                        stroke="none"
                      />
                    </svg>
                    <span
                      class="recharts-legend-item-text"
                    >
                      Republican
                    </span>
                  </li>
                  <li
                    class="recharts-legend-item legend-item-2"
                    style="display: inline-block; margin-right: 10px;"
                  >
                    <svg
                      class="recharts-surface"
                      height="14"
                      style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                      version="1.1"
                      viewBox="0 0 32 32"
                      width="14"
                    >
                      <path
                        class="recharts-legend-icon"
                        d="M0,4h32v24h-32z"
                        fill="#00D136"
                        stroke="none"
                      />
                    </svg>
                    <span
                      class="recharts-legend-item-text"
                    >
                      Male
                    </span>
                  </li>
                  <li
                    class="recharts-legend-item legend-item-3"
                    style="display: inline-block; margin-right: 10px;"
                  >
                    <svg
                      class="recharts-surface"
                      height="14"
                      style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                      version="1.1"
                      viewBox="0 0 32 32"
                      width="14"
                    >
                      <path
                        class="recharts-legend-icon"
                        d="M0,4h32v24h-32z"
                        fill="#B533FF"
                        stroke="none"
                      />
                    </svg>
                    <span
                      class="recharts-legend-item-text"
                    >
                      Female
                    </span>
                  </li>
                </ul>
              </div>
              <div
                class="recharts-tooltip-wrapper"
                style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
              >
                <div
                  class="recharts-default-tooltip"
                  style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
                >
                  <p
                    class="recharts-tooltip-label"
                    style="margin: 0px;"
                  />
                </div>
              </div>
            </div>
          </section>
          <section
            class="chamber-data"
          >
            <section>
              <h4>
                Average Age
              </h4>
              <div>
                
                <span>
                  58.52
                   years old
                </span>
              </div>
              <section>
                <h5>
                  By Party
                </h5>
                <div>
                  <label>
                    Democrat
                    : 
                  </label>
                  <span>
                    59.39
                     years old
                  </span>
                </div>
                <div>
                  <label>
                    Republican
                    : 
                  </label>
                  <span>
                    57.62
                     years old
                  </span>
                </div>
              </section>
              <section>
                <h5>
                  By Gender
                </h5>
                <div>
                  <label>
                    Female
                    : 
                  </label>
                  <span>
                    58.93
                     years old
                  </span>
                </div>
                <div>
                  <label>
                    Male
                    : 
                  </label>
                  <span>
                    58.4
                     years old
                  </span>
                </div>
              </section>
            </section>
            <section>
              <h4>
                Youngest
                 Member
              </h4>
              <label>
                Alexandria Ocasio-Cortez
              </label>
              <div>
                <label>
                  Age
                  : 
                </label>
                <span>
                  30
                   years old
                </span>
              </div>
              <div>
                <label>
                  Date of Birth
                  : 
                </label>
                <span>
                  1989-10-13
                </span>
              </div>
              <div>
                <label>
                  State Representation
                  : 
                </label>
                <span>
                  NY
                </span>
              </div>
              <div>
                <label>
                  Gender
                  : 
                </label>
                <span>
                  Female
                </span>
              </div>
              <div>
                <label>
                  Party
                  : 
                </label>
                <span>
                  Democrat
                </span>
              </div>
            </section>
            <section>
              <h4>
                Oldest
                 Member
              </h4>
              <label>
                Don Young
              </label>
              <div>
                <label>
                  Age
                  : 
                </label>
                <span>
                  87
                   years old
                </span>
              </div>
              <div>
                <label>
                  Date of Birth
                  : 
                </label>
                <span>
                  1933-06-09
                </span>
              </div>
              <div>
                <label>
                  State Representation
                  : 
                </label>
                <span>
                  AK
                </span>
              </div>
              <div>
                <label>
                  Gender
                  : 
                </label>
                <span>
                  Male
                </span>
              </div>
              <div>
                <label>
                  Party
                  : 
                </label>
                <span>
                  Republican
                </span>
              </div>
            </section>
            <section>
              <h4>
                Number of Members by Gender
              </h4>
              <div>
                <label>
                  Total
                  : 
                </label>
                <span>
                  449
                </span>
              </div>
              <div>
                <label>
                  Female
                  : 
                </label>
                <span>
                  106 (23.61%)
                </span>
              </div>
              <div>
                <label>
                  Male
                  : 
                </label>
                <span>
                  343 (76.39%)
                </span>
              </div>
            </section>
            <section>
              <h4>
                Number of Members by Party
              </h4>
              <h5>
                Democrat
              </h5>
              <div>
                <label>
                  Total
                  : 
                </label>
                <span>
                  239
                </span>
              </div>
              <div>
                <label>
                  Female
                  : 
                </label>
                <span>
                  91 (38.08%)
                </span>
              </div>
              <div>
                <label>
                  Male
                  : 
                </label>
                <span>
                  148 (61.92%)
                </span>
              </div>
              <h5>
                Republican
              </h5>
              <div>
                <label>
                  Total
                  : 
                </label>
                <span>
                  209
                </span>
              </div>
              <div>
                <label>
                  Female
                  : 
                </label>
                <span>
                  15 (7.18%)
                </span>
              </div>
              <div>
                <label>
                  Male
                  : 
                </label>
                <span>
                  194 (92.82%)
                </span>
              </div>
            </section>
          </section>
        </section>
      </article>
      <article>
        <h3>
          Senate
        </h3>
        <hr />
        <section
          class="chamber"
        >
          <section
            class="chamber-data"
          >
            <section>
              <h4>
                Average Age
              </h4>
              <div>
                
                <span>
                  63.89
                   years old
                </span>
              </div>
              <section>
                <h5>
                  By Party
                </h5>
                <div>
                  <label>
                    Democrat
                    : 
                  </label>
                  <span>
                    63.87
                     years old
                  </span>
                </div>
                <div>
                  <label>
                    Republican
                    : 
                  </label>
                  <span>
                    63.43
                     years old
                  </span>
                </div>
              </section>
              <section>
                <h5>
                  By Gender
                </h5>
                <div>
                  <label>
                    Female
                    : 
                  </label>
                  <span>
                    62.08
                     years old
                  </span>
                </div>
                <div>
                  <label>
                    Male
                    : 
                  </label>
                  <span>
                    64.52
                     years old
                  </span>
                </div>
              </section>
            </section>
            <section>
              <h4>
                Youngest
                 Member
              </h4>
              <label>
                Joshua Hawley
              </label>
              <div>
                <label>
                  Age
                  : 
                </label>
                <span>
                  40
                   years old
                </span>
              </div>
              <div>
                <label>
                  Date of Birth
                  : 
                </label>
                <span>
                  1979-12-31
                </span>
              </div>
              <div>
                <label>
                  State Representation
                  : 
                </label>
                <span>
                  MO
                </span>
              </div>
              <div>
                <label>
                  Gender
                  : 
                </label>
                <span>
                  Male
                </span>
              </div>
              <div>
                <label>
                  Party
                  : 
                </label>
                <span>
                  Republican
                </span>
              </div>
            </section>
            <section>
              <h4>
                Oldest
                 Member
              </h4>
              <label>
                Dianne Feinstein
              </label>
              <div>
                <label>
                  Age
                  : 
                </label>
                <span>
                  87
                   years old
                </span>
              </div>
              <div>
                <label>
                  Date of Birth
                  : 
                </label>
                <span>
                  1933-06-22
                </span>
              </div>
              <div>
                <label>
                  State Representation
                  : 
                </label>
                <span>
                  CA
                </span>
              </div>
              <div>
                <label>
                  Gender
                  : 
                </label>
                <span>
                  Female
                </span>
              </div>
              <div>
                <label>
                  Party
                  : 
                </label>
                <span>
                  Democrat
                </span>
              </div>
            </section>
            <section>
              <h4>
                Number of Members by Gender
              </h4>
              <div>
                <label>
                  Total
                  : 
                </label>
                <span>
                  101
                </span>
              </div>
              <div>
                <label>
                  Female
                  : 
                </label>
                <span>
                  26 (25.74%)
                </span>
              </div>
              <div>
                <label>
                  Male
                  : 
                </label>
                <span>
                  75 (74.26%)
                </span>
              </div>
            </section>
            <section>
              <h4>
                Number of Members by Party
              </h4>
              <h5>
                Democrat
              </h5>
              <div>
                <label>
                  Total
                  : 
                </label>
                <span>
                  45
                </span>
              </div>
              <div>
                <label>
                  Female
                  : 
                </label>
                <span>
                  17 (37.78%)
                </span>
              </div>
              <div>
                <label>
                  Male
                  : 
                </label>
                <span>
                  28 (62.22%)
                </span>
              </div>
              <h5>
                Republican
              </h5>
              <div>
                <label>
                  Total
                  : 
                </label>
                <span>
                  54
                </span>
              </div>
              <div>
                <label>
                  Female
                  : 
                </label>
                <span>
                  9 (16.67%)
                </span>
              </div>
              <div>
                <label>
                  Male
                  : 
                </label>
                <span>
                  45 (83.33%)
                </span>
              </div>
            </section>
          </section>
        </section>
      </article>
    </div>
  </section>
</div>
`;

exports[`Recipes Component when \`useFetch\` returns data when 500 status returned displays error messaging 1`] = `
<div>
  <section>
    <h2
      class="congress-header"
    >
      Congressional Information
    </h2>
    <section>
      <h3>
        Whoops! There was a problem loading
         
        Congress data.
      </h3>
      <p>
        Please reload browser to try again in a little bit.
      </p>
    </section>
  </section>
</div>
`;
